packages/core/src/lib/favorite/favorite.service.ts

```
@@ -71,14 +76,14 @@ export class FavoriteService extends TenantAwareCrudService<Favorite> {
-				favorite = new Favorite(entity);
+				favorite = new Favorite({ ...entity, employeeId });
 			}
 
-			// If favorite element not exists, create and return new one
+			// Create or return the existing favorite
 			return await this.save(favorite);
 		} catch (error) {
-			console.log(error);
-			throw new BadRequestException('Favorite creation failed', error);
+			console.error('Error while creating favorite:', error);
+			throw new BadRequestException('Favorite creation failed: ' + (error?.message || error));
```

Biome (1.9.4)

[error] Template literals are preferred over string concatenation.

Unsafe fix: Use a template literal.


(lint/style/useTemplate)